kin-backup-and-restore/kin-backup-and-restore/src/main/java/kin/backupandrestore/backup/presenter/CreatePasswordPresenterImpl.java

```diff
@@ -41,55 +41,41 @@ public void onBackClicked() {
 	}
 
 	@Override
-	public void enterPasswordChanged(String password, String confirmPassword) {
-		if (validatePassword(password)) {
+	public void passwordChanged(String changedPassword, String otherPassword, boolean isConfirmPassword) {
+		boolean changedPasswordIsEmpty = changedPassword.isEmpty();
+		if (validatePassword(changedPassword)) {
 			isPasswordRulesOK = true;
-			if (view != null) {
-				view.setEnterPasswordIsCorrect(true);
-			}
-			checkConfirmPassword(password, confirmPassword);
+			handlePasswordCorrectness(isConfirmPassword, true);
 		} else {
 			isPasswordRulesOK = false;
-			if (password.isEmpty()) {
-				if (view != null) {
-					view.resetEnterPasswordField();
-					view.resetConfirmPasswordField();
-				}
+			if (changedPasswordIsEmpty) {
+				handlePasswordIsEmpty(isConfirmPassword);
 			} else {
-				if (view != null) {
-					view.setEnterPasswordIsCorrect(false);
-				}
-				checkConfirmPassword(password, confirmPassword);
+				handlePasswordCorrectness(isConfirmPassword, false);
 			}
 		}
+		isOnOfPasswordsEmpty = otherPassword.isEmpty() || changedPasswordIsEmpty;
 		checkAllCompleted();
 	}
 
-	private void checkConfirmPassword(String password, String confirmPassword) {
-		if (!password.isEmpty() && !confirmPassword.isEmpty()) {
-			if (password.equals(confirmPassword)) {
-				isPasswordsMatches = true;
-				if (view != null) {
-					view.setConfirmPasswordIsCorrect(true);
-					view.closeKeyboard();
-				}
+	private void handlePasswordCorrectness(boolean isConfirmPassword, boolean isCorrect) {
+		if (view != null) {
```

Comment:
In my opinion it is better to check the password after each letter that was entered. It does not make sense that the user will fill the correct password but will not get any indication for that.
@mrsegev